<launch>

  <arg name="sim"             default="false" />
  <arg name="camera"          default="sr300" />

  <param name="use_sim_time" value="false" />

  <node name="stack_each" pkg="grasp_and_place" type="stack_each_node" respawn="false" output="screen">
    <rosparam command="load" file="$(find grasp_and_place)/config/stack_each_parameters.yaml"/>
    <rosparam command="load" file="$(find grasp_and_place)/config/grasping_poses.yaml"/>
    <rosparam command="load" file="$(find grasp_and_place)/config/collision_environment.yaml"/>
    <param name="sim" value="$(arg sim)" />
  </node>

  <include file="$(find object_detection)/launch/frame_broadcaster.launch" />
  <include file="$(find realsense_camera)/launch/sr300_nodelet_rgbd.launch">
    <arg name="camera"        value="$(arg camera)"/>
  </include>

  <node pkg="object_detection" type="object_detection_service" name="object_detection_service" output="screen">
   	<rosparam command="load" file="$(find object_detection)/params/detection.yaml"/>
  </node>

  <node pkg="object_detection" type="object_localisation_service" name="object_localisation_service" output="screen">
   	<rosparam command="load" file="$(find object_detection)/params/localisation.yaml"/>
  </node>

  <node pkg="object_detection" type="object_pose_refinement_service" name="object_pose_refinement_service" output="screen">
   <rosparam command="load" file="$(find object_detection)/params/refinement.yaml"/>
  </node>

  <group if="$(arg sim)">
  	<include file="$(find urdf_models)/launch/stacking_stones.launch"/>
  </group>
  
  <!-- <include file="$(find gazebo_stack_pose_planner)/launch/online_stacking.launch" /> -->
  

</launch>
